src/util/TypeTraits.h

```diff
@@ -215,9 +215,24 @@ template <typename... Ts>
 requires(sizeof...(Ts) > 0)
 using First = typename detail::FirstWrapper<Ts...>::type;
 
+/*
+The difference between the following concepts, for checking if types are
+invocable with given types and what their return type looks like, and
+`std::is_invocable_r_v` is, that `std::is_invocable_r_v` only checks, if the
+actual return type can be converted to the wanted return type.
```

Comment:
how about:
"The following concepts are similar to `std::is_invocable_r_v` with the following difference: `std::is_invocable_r_v` only checks that the return type of a function is convertible to the specified type, but the following concepts check for an exact match of the return type (`InvocableWithExactReturnType`) or a similar return type (meaning that the types are the same when ignoring `const`, `volatile`, and reference qualifiers, `InvocableWithSimilarReturnType`.

Reply:
I've madesome slight adjustments, see [74769b4](https://github.com/ad-freiburg/qlever/pull/1118/commits/74769b419c5a9e2c31a9c17f26705b0f35f43bcd).